pkg/validation/app.go

```diff
@@ -367,7 +367,8 @@ func (v *Validator) validateMetadataConstraints(ctx context.Context, cr v1alpha1
 
 	for _, app := range apps {
 		if app.Spec.Name == cr.Spec.Name {
-			if entry.Spec.Restrictions.ClusterSingleton {
+			if entry.Spec.Restrictions.ClusterSingleton &&
+				app.Spec.KubeConfig.Context.Name == cr.Spec.KubeConfig.Context.Name {
```

Comment:
Why do we need this?

Reply:
prometheus-operator-crd is clustersingleton. 
and it gets denied because in same NS (org) on CAPI.

https://github.com/giantswarm/giantswarm/issues/24246#issuecomment-1315138367

Reply:
No it gets denied because the default-apps is installed on the MC also so we have the prometheus-operator-crd from flux and one from the defautl apps

Reply:
let's pair on it after the training

Reply:
from the code of validation, it checks for the `app.spec.name` which is the same: 
https://github.com/giantswarm/app/blob/c262c142bd0accf676e65a64a402f277800f96fd/pkg/validation/app.go#L369

Reply:
Sure, but the issue is that the app name should start with the cluster_id (i.e. med01-prometheus-operator-crd) but it is not

Reply:
it is )

Reply:
Then let's pair :D

Reply:
the log saying 
```
 `prometheus-operator-crd` can only be installed once in cluster `org-multi-project`
```

it assumes the org as a namespace.

Reply:
True